[DCP] Add tests to demonstrate DCP checkpoint conversion

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117773
* #117772
* #116248

As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @LucasLLC

fegin committed:
[DCP][RFC] Add tests to demonstrate DCP checkpoint conversion

As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

[ghstack-poisoned]

fegin committed:
Update on "[DCP][RFC] Add tests to demonstrate DCP checkpoint conversion"


As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

fegin committed:
Update on "[DCP][RFC] Add tests to demonstrate DCP checkpoint conversion"


As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

wconstab commented on test/distributed/checkpoint/test_torch_save_to_dcp.py:
```diff
@@ -0,0 +1,248 @@
+        return x
+
+
+def _construct_sharded_state_dict(fp_name: str, world_size: int) -> Dict[str, Any]:
```
seems odd to me that this util wouldn't already exist somewhere in the real dcp code or as a torch util.  should it be one (and any opportunity to replace existing code with it?)

fegin replied:
Yup, I agree this should be an util in DCP.

wconstab commented on test/distributed/checkpoint/test_torch_save_to_dcp.py:
```diff
@@ -0,0 +1,248 @@
+                planner.commit_tensor(req, target_tensor)
+
+        fut: Future = Future()
+        fut.set_result(None)
```
strange use of future?

fegin replied:
This is because `read_data` requires a future. So we simply create one and assign a None to it. I'm not sure why the original intention to have a `Future[None]` as the return value.

fegin committed:
Update on "[DCP][RFC] Add tests to demonstrate DCP checkpoint conversion"


As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

fegin committed:
Update on "[DCP] Add tests to demonstrate DCP checkpoint conversion"


As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

fegin committed:
Update on "[DCP] Add tests to demonstrate DCP checkpoint conversion"


As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

fegin committed:
Update on "[DCP] Add tests to demonstrate DCP checkpoint conversion"


As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

fegin committed:
Update on "[DCP] Add tests to demonstrate DCP checkpoint conversion"


As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

fegin committed:
Update on "[DCP] Add tests to demonstrate DCP checkpoint conversion"


As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

fegin committed:
Update on "[DCP] Add tests to demonstrate DCP checkpoint conversion"


As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

fegin committed:
Update on "[DCP] Add tests to demonstrate DCP checkpoint conversion"


As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

fegin committed:
Update on "[DCP] Add tests to demonstrate DCP checkpoint conversion"


As title

Differential Revision: [D52854759](https://our.internmc.facebook.com/intern/diff/D52854759/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]
